### PR TITLE
Simplify base inheritance statement with Mojo::Base everywhere

### PR DESCRIPTION
**Pull request by okurz**
Based on
* #1913
* #1926
* #1789

The patch coverage will be still be our automatic acceptance threshold due to background processes not collecting coverage properly. If you can approve in the current state please approve which I will see as implicit approval of the other three PRs, otherwise please review the PRs above in the stated order.

EDIT: I managed to exclude the consoles::VNC changes which cause the codecov patch coverage check to fail so this PR now only includes the one commit about "base inheritance using Mojo::Base"

### CODE REVIEW

**Comment by Martchus** [comment]
It would be good if @kraih could have a look (although at the moment this is only build on top of several other changes which should likely be merged before).

**Review by kraih — APPROVED** [review]
It's a large patch, so easy to overlook something. But it seems fine to me.

File: consoles/VNC.pm [hunk 1]
```diff
@@ -1093,13 +1030,11 @@ sub _receive_colour_map ($self) {
     $self->socket->read(my $map_infos, 5);
     my ($padding, $first_colour, $number_of_colours) = unpack('Cnn', $map_infos);
 
-    for (my $i = 0; $i < $number_of_colours; $i++) {
+    for (0 .. $number_of_colours) {
         $self->socket->read(my $colour, 6);
         my ($red, $green, $blue) = unpack('nnn', $colour);
-        tinycv::set_colour($self->vncinfo, $first_colour + $i, $red / 256, $green / 256, $blue / 256);
+        tinycv::set_colour($self->vncinfo, $first_colour + $_, $red / 256, $green / 256, $blue / 256);
```
Comment (perlpunk) [inline consoles/VNC.pm:1033]: this includes `$number_of_colours`. before it was `$i < $number_of_colours`.

**Review by perlpunk — CHANGES_REQUESTED** [review]

File: consoles/vnc_base.pm [hunk 2]
```diff
@@ -178,26 +171,15 @@ sub mouse_hide ($self, $args) {
...
-        $mask = $bstate;
-    }
-    elsif ($button eq 'right') {
-        $mask = $bstate << 2;
-    }
-    elsif ($button eq 'middle') {
-        $mask = $bstate << 1;
-    }
+    my %mask = (left => $bstate, right => $bstate << 2, middle => $bstate << 1);
+    my $mask = $mask{button} // 0;
```
Comment (perlpunk) [inline consoles/vnc_base.pm:179]: `$mask{button}` misses a `$`.
```suggestion
    my $mask = {left => $bstate, right => $bstate << 2, middle => $bstate << 1}->{$button} // 0;
```

File: consoles/serial_screen.pm [hunk 3]
```diff
@@ -155,9 +147,7 @@ An undefined timeout will cause to wait indefinitely. A timeout of 0 means to
 just read once.
 
 =cut
-sub do_read
-{
-    my ($self, undef, %args) = @_;
+sub do_read ($self, $buf, %args) {
```
Comment (perlpunk) [inline consoles/serial_screen.pm:125]:
```suggestion
sub do_read ($self, $, %args) {
```
like in `consoles/ssh_screen.pm`

File: consoles/vnc_base.pm [hunk 4]
```diff
@@ -52,17 +46,15 @@ sub connect_remote {
     return $self->{vnc};
 }
 
-sub request_screen_update {
-    my ($self) = @_;
+sub request_screen_update ($self, @) {
```
Comment (perlpunk) [inline consoles/vnc_base.pm:39]: is there a case where the method is called with an argument?

**Comment by perlpunk** [comment]
> Based on #1913 and #1789

Is there a reason that you created this PR based on two other PRs?
Are you expecting people to review this PR although it contains the same commits as other PRs?
Which one should people review?
Mark other PRs as draft or try to help in any other way that people don't have double work, please.

File: consoles/vnc_base.pm (hunk 4 above)
Comment (okurz) [reply consoles/vnc_base.pm:39]: I am not exactly sure but I think so when used as a callback target over the RPC implementation which somehow ends up with more unused, unfilled arguments passed. I might have overdone it in one or two cases but normally I would only add the argument if some tests trigger an error or warning

**Comment by okurz** [comment]
> > Based on #1913 and #1789
> 
> Is there a reason that you created this PR based on two other PRs?

Yes, because otherwise changes would trigger conflicts. I could invest the additional effort to split out commits ito separate PRs for easier review. As soon as one PR is merged the other will show up as conflicting and can not be merged so I would need to rebase and fix the conflict and upload again which you or others might see as need to review again so it's all additional effort that I would like to avoid if I can.

> Are you expecting people to review this PR although it contains the same commits as other PRs?

Well, I hoped that you would have seen the description so that it's clear which commits are unique to this PR. Sorry, I would like if we could make this more clear in github somehow.

> Which one should people review? Mark other PRs as draft or try to help in any other way that people don't have double work, please.

Yes, I will research if there are better solutions. I did mark all PRs as ready for review because in the best case there would have been no problems and in that case there would not be any more work necessary.

File: consoles/serial_screen.pm (hunk 3 above)
Comment (okurz) [reply consoles/serial_screen.pm]: done

File: consoles/vnc_base.pm (hunk 2 above)
Comment (okurz) [reply consoles/vnc_base.pm]: done, interesting syntax but makes sense and looks nice :)

File: consoles/VNC.pm (hunk 1 above)
Comment (okurz) [reply consoles/VNC.pm:1033]: right, fixed. thx

File: consoles/VNC.pm [hunk 5]
```diff
@@ -1139,38 +1023,30 @@ sub _receive_ikvm_encoding {
...
 
-sub _receive_colour_map {
-    my $self = shift;
-
+sub _receive_colour_map ($self) {
     $self->socket->read(my $map_infos, 5);
     my ($padding, $first_colour, $number_of_colours) = unpack('Cnn', $map_infos);
 
-    for (my $i = 0; $i < $number_of_colours; $i++) {
+    for (0 .. $number_of_colours - 1) {
```
Comment (kraih) [inline consoles/VNC.pm:1030]: `for my $i (0 ...` is cleaner than relying on `$_`.

Comment (okurz) [reply consoles/VNC.pm]: done

**Review by kraih — APPROVED** [review]

**Comment by Martchus** [comment]
I must say I haven't dared to approve the VNC changes yet (since the test coverage is unclear).

**Comment by kraih** [comment]
Better test coverage would be nice indeed. But the changes themselves are not very risky and easy enough to revert if necessary.

**Comment by okurz** [comment]
I managed to exclude the consoles::VNC changes which cause the codecov patch coverage check to fail so this PR now only includes the one commit about "base inheritance using Mojo::Base". Let's see what the codecov check says now.

**Review by baierjan — APPROVED** [review]

**Pull request merged**